use-cases/using-SPDK-vhostuser-and-kata.md

```diff
@@ -0,0 +1,153 @@
+# Setup to run SPDK vhost-user devices with Kata Containers and Docker*
+
+- [SPDK vhost-user target overview](#spdk-vhost-user-target-overview)
+- [Install and setup SPDK vhost-user target](#install-and-setup-spdk-vhost-user-target)
+  - [Get source code and build SPDK](#get-source-code-and-build-spdk)
+  - [Run SPDK vhost-user target](#run-spdk-vhost-user-target)
+- [Host setup for vhost-user devices](#host-setup-for-vhost-user-devices)
+- [Launch a Kata container with SPDK vhost-user block device](#launch-a-kata-container-with-spdk-vhost-user-block-device)
+
+## SPDK vhost-user Target Overview
+
+The Storage Performance Development Kit (SPDK) provides a set of tools and
+libraries for writing high performance, scalable, user-mode storage applications.
+
+While, virtio is a efficient way to transport data for virtual environments and
+guests. It is most commonly used in QEMU VMs, where the QEMU itself exposes a
+virtual PCI device and the guest OS communicates with it using a specific Virtio
+PCI driver.
+vhost is a protocol for devices accessible via inter-process communication. It
+uses the same virtio queue layout as Virtio to allow vhost devices to be mapped
+directly to virtio devices. The initial vhost implementation is a part of the
+Linux kernel and uses ioctl interface to communicate with userspace applications.
+vhost-user implements the control plane through Unix domain socket to establish
+virtio queue sharing with a user space process on the same host. So SPDK can expose
+vhost device via vhost-user protocol.
+
+SPDK vhost is a vhost-user slave server. It exposes Unix domain sockets and
+allows external applications to connect. It is capable of exposing virtualized
+storage devices to QEMU instances or other arbitrary processes.
+
+Currently, SPDK vhost-user target can exposes these types of virtualized devices:
+
+- `vhost-user-blk`
+- `vhost-user-scsi`
+- `vhost-user-nvme`
+
+For more information, visit [SPDK](https://spdk.io) and [SPDK vhost-user target](https://spdk.io/doc/vhost.html).
+
+## Install and setup SPDK vhost-user target
+
+### Get source code and build SPDK
+
+Following SPDK [getting started](https://spdk.io/doc/getting_started.html)
+
+### Run SPDK vhost-user target
+
+First, run the SPDK setup.sh script to setup some hugepages for the SPDK vhost
+target application. This will allocate 4096MiB (4GiB) of hugepages, enough for
+the SPDK vhost target and the virtual machine.
+
+```bash
+$ sudo HUGEMEM=4096 scripts/setup.sh
+```
+
+Then, take directory `/var/run/kata-containers/vhost-user` as Kata's vhost-user
+device directory. Make subdirectories for vhost-user sockets and device nodes.
+
+For more details, see section [Host setup for vhost-user devices](#host-setup-for-vhost-user-devices).
+
+```bash
+mkdir -p /var/run/kata-containers/vhost-user/
```

Comment:
- I think you're missing a `sudo` at the start of each of these calls. Which reminds me...

- Ideally, we like our documents to be "executable" meaning that if you can write them such that the CI can run all `bash` blocks, we could "run" this document to ensure it is always correct (I hate incorrect documentation that's "gone stale" :). Do you think this is realistic for SPDK?

Reply:
Ack.

This is realistic for SPDK :)
I just recap all the bash blocks. SPDK specific commands are：
- scripts/setup.sh
- app/spdk_tgt/spdk_tgt
- RPC bdev_malloc_create and vhost_create_blk_controller
setup.sh and spdk_tgt are always the stable ones, SPDK community has no reason and no plan to change their usage models.
Previously, SPDK didn't  make a rule for RPC method naming. So in last releases, we established a formal naming method for all SPDK RPC methods. bdev_malloc_create and vhost_create_blk_controller RPC methods are already well standardized and permanent.
All of these SPDK commands are not temporary.